### PR TITLE
boards: adapt LVGL v8 defconfig

### PR DESCRIPTION
**Pull request by FASTSHIFT**
## Summary

Adapt to LVGL v8 update: https://github.com/apache/incubator-nuttx-apps/pull/1341

## Impact

LVGL related configuration may not work properly.

```bash
imxrt1064-evk:lvgl
imxrt1060-evk:lvgl
stm32f746g-disco:lvgl
lpcxpresso-lpc54628:lvgl
stm32f429i-disco:lvgl
esp32c3-devkit:lvgl
ttgo_t_display_esp32:lvgl_fb
ttgo_t_display_esp32:lvgl_lcd
esp32-wrover-kit:lvgl
sim:lvgl_fb
sim:lvgl_lcd
```

### CODE REVIEW

File: boards/xtensa/esp32/esp32-wrover-kit/configs/lvgl/defconfig [hunk 1]
```diff
@@ -42,7 +42,13 @@ CONFIG_LCD_DEV=y
 CONFIG_LCD_ILI9341=y
 CONFIG_LCD_ILI9341_IFACE0=y
 CONFIG_LV_COLOR_16_SWAP=y
-CONFIG_LV_DPI=125
+CONFIG_LV_MEM_CUSTOM=y
+CONFIG_LV_PORT_USE_LCDDEV=y
+CONFIG_LV_PORT_USE_TOUCHPAD=y
```
Comment (gustavonihei) [inline boards/xtensa/esp32/esp32-wrover-kit/configs/lvgl/defconfig:47]: ESP-WROVER-KIT does not provide input support with LVGL.

Comment (gustavonihei) [reply boards/xtensa/esp32/esp32-wrover-kit/configs/lvgl/defconfig:47]: Sure, I am testing it right now.
But so far `lvgldemo widgets` is failing to run:
```
nsh> lvgldemo widgets
[LVGL] [Warn]   (5.340, +5340)   lv_demo_widgets: LV_FONT_MONTSERRAT_18 is not enabled for the widgets demo. Using LV_FONT_DEFAULT instead.     (in lv_demo_widgets.c line #138)
[LVGL] [Warn]   (5.340, +0)      lv_demo_widgets: LV_FONT_MONTSERRAT_12 is not enabled for the widgets demo. Using LV_FONT_DEFAULT instead.     (in lv_demo_widgets.c line #143)
[LVGL] [Error]  (5.860, +520)    lcddev_update_thread: ioctl(LCDDEVIO_PUTAREA) failed: 22       (in lv_port_lcddev.c line #104)
```

Comment (gustavonihei) [reply boards/xtensa/esp32/esp32-wrover-kit/configs/lvgl/defconfig:47]: With the patch from #7277, I managed to execute the Widgets demo on the ESP-WROVER-KIT development board.

Comment (gustavonihei) [reply boards/xtensa/esp32/esp32-wrover-kit/configs/lvgl/defconfig:47]: `CONFIG_LV_PORT_USE_TOUCHPAD` still needs to be removed from the defconfig.

**Comment by FASTSHIFT** [comment]
@gustavonihei 
Can you help test other ESP32 boards? :)

**Comment by gustavonihei** [comment]
> @gustavonihei Can you help test other ESP32 boards? :)

I have just the ESP-WROVER-KIT here, I believe @tmedicci could help testing with the `ttgo_t_display_esp32` platform.

File: boards/xtensa/esp32/ttgo_t_display_esp32/configs/lvgl_lcd/defconfig [hunk 2]
```diff
@@ -42,7 +42,13 @@ CONFIG_LCD_ST7789_XOFFSET=53
 CONFIG_LCD_ST7789_XRES=135
 CONFIG_LCD_ST7789_YOFFSET=40
 CONFIG_LCD_ST7789_YRES=240
-CONFIG_LV_HOR_RES=135
+CONFIG_LV_DEMO_WIDGETS_SLIDESHOW=y
```
Comment (tmedicci) [inline boards/xtensa/esp32/ttgo_t_display_esp32/configs/lvgl_lcd/defconfig:45]: Please, add `CONFIG_LV_MEM_CUSTOM` from the `menuconfig`.

**Comment by tmedicci** [comment]
> > @gustavonihei Can you help test other ESP32 boards? :)
> 
> I have just the ESP-WROVER-KIT here, I believe @tmedicci could help testing with the `ttgo_t_display_esp32` platform.

I was able to test on _ttgo_t_display_esp32_. `ttgo_t_display_esp32:lvgl_fb` worked successfully. `ttgo_t_display_esp32:lvgl_lcd` fails to build because it lacks a parameters I've commented on code.

File: boards/xtensa/esp32/ttgo_t_display_esp32/configs/lvgl_lcd/defconfig (hunk 2 above)
Comment (tmedicci) [inline boards/xtensa/esp32/ttgo_t_display_esp32/configs/lvgl_lcd/defconfig:45]: I think the slideshow example was removed, so that could be removed.

Comment (tmedicci) [inline boards/xtensa/esp32/ttgo_t_display_esp32/configs/lvgl_lcd/defconfig:45]: Please, add (from `menuconfig`) the following flags:
1. `CONFIG_LV_FONT_MONTSERRAT_12`
2. `CONFIG_LV_FONT_MONTSERRAT_18`

File: boards/xtensa/esp32/ttgo_t_display_esp32/configs/lvgl_fb/defconfig [hunk 3]
```diff
@@ -43,7 +43,13 @@ CONFIG_LCD_ST7789_XOFFSET=53
 CONFIG_LCD_ST7789_XRES=135
 CONFIG_LCD_ST7789_YOFFSET=40
 CONFIG_LCD_ST7789_YRES=240
-CONFIG_LV_HOR_RES=135
+CONFIG_LV_DEMO_WIDGETS_SLIDESHOW=y
+CONFIG_LV_MEM_CUSTOM=y
```
Comment (tmedicci) [inline boards/xtensa/esp32/ttgo_t_display_esp32/configs/lvgl_fb/defconfig:47]: Please, add (from `menuconfig`) the following flags:
1. `CONFIG_LV_FONT_MONTSERRAT_12`
2. `CONFIG_LV_FONT_MONTSERRAT_18`

File: boards/xtensa/esp32/ttgo_t_display_esp32/configs/lvgl_fb/defconfig [hunk 4]
```diff
@@ -43,7 +43,13 @@ CONFIG_LCD_ST7789_XOFFSET=53
 CONFIG_LCD_ST7789_XRES=135
 CONFIG_LCD_ST7789_YOFFSET=40
 CONFIG_LCD_ST7789_YRES=240
-CONFIG_LV_HOR_RES=135
+CONFIG_LV_DEMO_WIDGETS_SLIDESHOW=y
```
Comment (tmedicci) [inline boards/xtensa/esp32/ttgo_t_display_esp32/configs/lvgl_fb/defconfig:46]: I think the slideshow example was removed, so that could be removed.

**Review by tmedicci — CHANGES_REQUESTED** [review]

**Comment by FASTSHIFT** [comment]
@tmedicci @gustavonihei 
Is there anything else we need to modify? Can we start merging this PR :)

**Review by gustavonihei — APPROVED** [review]

**Comment by tmedicci** [comment]
> @tmedicci @gustavonihei Is there anything else we need to modify? Can we start merging this PR :)

It isn't a block from my side, but I still see a warning regarding the framebuffer usage, please check https://github.com/apache/incubator-nuttx-apps/pull/1341/files#diff-2017a1909697a8fc95f0082144b1c6ef91522530ba90c8cd37caa318a5a346ec

I'd say that it's important to double-check the implementation.

**Comment by FASTSHIFT** [comment]
> > @tmedicci @gustavonihei Is there anything else we need to modify? Can we start merging this PR :)
> 
> It isn't a block from my side, but I still see a warning regarding the framebuffer usage, please check my last comment on [apache/nuttx-apps#1341](https://github.com/apache/nuttx-apps/pull/1341)
> 
> I'd say that it's important to double-check the implementation.

Sorry it took so long to reply.
Is the framebuffer warning `"fbmem2 is incorrect"` this one? I changed this to `INFO` level because this is designed to accommodate dual framebuffers with non-contiguous addresses. For single framebuffer devices, this is not required.

`CONFIG_LV_FONT_MONTSERRAT_12`
`CONFIG_LV_FONT_MONTSERRAT_18`
I don't think it's necessary, LVGL will automatically choose the default font, it doesn't affect the demo effect, turning off the non-necessary fonts can save FLASH space, I would like to turn them off again

**Comment by FASTSHIFT** [comment]
> > @tmedicci @gustavonihei Is there anything else we need to modify? Can we start merging this PR :)
> 
> It isn't a block from my side, but I still see a warning regarding the framebuffer usage, please check my last comment on [apache/nuttx-apps#1341](https://github.com/apache/nuttx-apps/pull/1341)
> 
> I'd say that it's important to double-check the implementation.

I don't seem to find any relevant comments on https://github.com/apache/nuttx-apps/pull/1341, can you try again :)

**Review by tmedicci — APPROVED** [review]
LGTM!

**Comment by FASTSHIFT** [comment]
@tmedicci 
Do you agree to turn off the font:
`CONFIG_LV_FONT_MONTSERRAT_12`
`CONFIG_LV_FONT_MONTSERRAT_18`

**Comment by tmedicci** [comment]
> @tmedicci Do you agree to turn off the font: `CONFIG_LV_FONT_MONTSERRAT_12` `CONFIG_LV_FONT_MONTSERRAT_18`

I see no problems turning off the fonts besides the warning. Particularly, I'd keep them enabled for that defconfig, but you can decide about it ;)

**Pull request merged**